File: iou-cordapp/src/main/java/com/atfx/sample/contract/IOUContract.java
package com.atfx.sample.contract;

import net.corda.core.contracts.CommandData;
import net.corda.core.contracts.Contract;
import net.corda.core.transactions.LedgerTransaction;
import org.jetbrains.annotations.NotNull;

public class IOUContract implements Contract {
    private final static String ID = "io.cryptoblk.sample.contract.IOUContract";

    @Override
    public void verify(@NotNull LedgerTransaction tx) throws IllegalArgumentException {

        if (!(tx.getOutputs().size() == 1)) {
            throw new IllegalArgumentException("Failed requirement: ???");
        }

        //if (!(tx.outputsOfType(IOUState.class.get)== 1)) {
        //    throw new IllegalArgumentException("Failed requirement: ???");
        //}
    }

    public interface Commands extends CommandData {
        class Create implements Commands {}
        class Update implements Commands {}
    }

}

File: iou-cordapp/src/main/kotlin/com/formssihk/sample/contract/IOUContract.kt
package com.formssihk.sample.contract

import com.formssihk.sample.state.IOUState
import net.corda.core.contracts.CommandData
import net.corda.core.contracts.Contract
import net.corda.core.contracts.Requirements.using
import net.corda.core.contracts.requireThat
import net.corda.core.transactions.LedgerTransaction

class IOUContract: Contract {
    companion object {
        @JvmStatic
        val ID = "io.cryptoblk.sample.contract.IOUContract"
    }

    /**
     * The verify() function of all the states' contracts must not throw an exception for a transaction to be
     * considered valid.
     */
    override fun verify(tx: LedgerTransaction) {
        //val command = tx.commands.requireSingleCommand<Commands.Create>()
        requireThat {
            // Generic constraints around the IOU transaction.
            //"No inputs should be consumed when issuing an IOU." using (tx.inputs.isEmpty())
            "Only one output state should be created." using (tx.outputs.size == 1)
            val out = tx.outputsOfType<IOUState>().single()
            //"All of the participants must be signers." using (command.signers.containsAll(out.participants.map { it.owningKey }))

        }
    }

    /**
     * This contract only implements one command, Create.
     */
    interface Commands : CommandData {
        class Create : Commands
        class Update : Commands
    }
}